appnav/src/main/kotlin/io/element/android/appnav/RootFlowNode.kt

```diff
@@ -259,9 +269,16 @@ class RootFlowNode @AssistedInject constructor(
     }
 
     private suspend fun attachSession(sessionId: SessionId): LoggedInFlowNode {
-        val cacheIndex = cacheService.cacheIndex().first()
-        return attachChild {
-            backstack.newRoot(NavTarget.LoggedInFlow(sessionId, cacheIndex))
+        //TODO handle multi-session
+        return waitForChildAttached { navTarget ->
+            navTarget is NavTarget.LoggedInFlow && navTarget.sessionId == sessionId
```

Comment:
nit: should this function be renamed? It sounds like it does the attaching, but now it just waits for a session to be attached elsewhere?

Reply:
I reused the name from appyx, because they already have a `waitForChildAttached` but without `navTarget` checking. 
It does not `attach` the node anymore, just waits to see it in the hierarchy. 
It was not a good thing to `attach` the node because it's already done by listening to the `isAuthenticatedFlow`